sdk/test-utils/perfstress/src/utils.ts

```diff
@@ -28,3 +30,40 @@ export async function drainStream(stream: NodeJS.ReadableStream) {
     stream.on("error", reject);
   });
 }
+export async function makeRequest(
+  uri: string,
+  requestOptions: RequestOptions
+): Promise<IncomingMessage> {
+  return new Promise<IncomingMessage>((resolve, reject) => {
+    const req = request(uri, requestOptions, resolve);
+
+    req.once("error", (err) => {
+      reject(new Error(err.message));
+    });
+
+    req.end();
+  });
+}
+
+const _cachedProxyClients: {
+  v1: TestProxyHttpClientV1 | undefined;
+  v2: TestProxyHttpClient | undefined;
+} = {
+  v1: undefined,
+  v2: undefined
+};
+
+export const CachedProxyClients = {
+  getHttpClientV1(url: string): TestProxyHttpClientV1 {
+    if (_cachedProxyClients.v1) return _cachedProxyClients.v1;
+    // Create a new client if not defined already
+    _cachedProxyClients.v1 = new TestProxyHttpClientV1(url);
+    return _cachedProxyClients.v1;
+  },
+
+  getHttpClient(url: string): TestProxyHttpClient {
+    if (_cachedProxyClients.v2) return _cachedProxyClients.v2;
+    _cachedProxyClients.v2 = new TestProxyHttpClient(url);
+    return _cachedProxyClients.v2;
+  }
+};
```

Comment:
style nit: no need for wrapper object, also I find it slightly cleaner to initialize the cache if necessary to avoid two return statements

```suggestion
  export function getHttpClientV1(url: string): TestProxyHttpClientV1 {
    if (!_cachedProxyClients.v1) {
      _cachedProxyClients.v1 = new TestProxyHttpClientV1(url);
    }
    
    return _cachedProxyClients.v1;
  }

  export function getHttpClient(url: string): TestProxyHttpClient {
    if (!_cachedProxyClients.v2) {
      _cachedProxyClients.v2 = new TestProxyHttpClient(url);
    }
    
    return _cachedProxyClients.v2;
  }
```